client/go/dogma/content_service.go

```diff
@@ -0,0 +1,316 @@
+// Copyright 2018 LINE Corporation
+//
+// LINE Corporation licenses this file to you under the Apache License,
+// version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at:
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+package dogma
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type contentService service
+
+// Query specifies a query on a file.
+type Query struct {
+	Path string
+	// QueryType can be "identity" or "json_path". "identity" is used to retrieve the content as it is.
+	// "json_path" applies a series of JSON path to the content.
+	// See https://github.com/json-path/JsonPath/blob/master/README.md
+	QueryType   string
+	Expressions []string
+}
+
+// Entry represents an entry in the repository.
+type Entry struct {
+	Path       string      `json:"path"`
+	Type       string      `json:"type"` // can be JSON, TEXT or DIRECTORY
```

Comment:
same with above comment about iota

Reply:
This field is used to store the JSON response. If I use this as `iota`, I need a custom deserializer  to convert the sting to corresponding iota number.
Can I implement the custom deserializer or is it better to use predefined string constants?

Reply:
As a user, I would prefer iota constant. IMHO it's fine to add deserialize function like `func (o *Type) UnmarshalJSON(b []byte) error {}`

Reply:
Got it. I will try the way you suggest. Thang you!